### ISSUE TITLE
Different fingerprints based on protocol

### ISSUE THREAD

**Issue by edi** [fingerprintjs/fingerprintjs2#458]
### Scenario

- What's the scenario, what happens and what did you expect to happen? 
http and https give different fingerprints.

- What device and browser are you using?
MBP 2018, it does not mater though, I've tried the same from a PC and a Laptop, both on windows, as well as on Chrome / Safari on macOS.

- What version of fingerprintjs are you using? (Bug reports not applicable to fingerprintjs master are subject to be closed without comment.)
Latest version.

### Fingerprint data

I've saved it on diffchecker ( as I had to figure out what's causing the difference ).
https://www.diffchecker.com/EUuRg5Mv

So, what happened is that `deviceMemory` is defined on `HTTPS` but `NOT_AVAILABLE` on
plain `HTTP`. Hence, I went on and excluded `deviceMemory`, which then caused `enumerateDevices` to show up. Note that the above result came from `default` configuration within `FingerprintJS`. So, after excluding the `deviceMemory`, `enumerateDevices` showed up, which wasn't there in the initial components list, as you can see in the diff checker.

`enumerateDevices` can be seen here, same, `https` on the left, `http` on the right: https://i.imgur.com/9fV4hJf.png, so, obviously I've excluded both of them, and now the fingerprint is consistent.

In conclusion, my question is, do you have any clue what else might be affected by generating the fingerprint in https vs http ? Just so I can disable those before going into production.

Thanks!

**Comment by jonashaag** [comment]
> `enumerateDevices` showed up, which wasn't there in the initial components list, as you can see in the diff checker.

That's because default exclude options aren't merged with your custom exclude options, rather, they are replaced. See #439.

> In conclusion, my question is, do you have any clue what else might be affected by generating the fingerprint in https vs http ? Just so I can disable those before going into production.

IIRC deviceMemory is the only HTTPS-only option we have. But not 100% sure. We should add to the docs whether an option is HTTPs dependent.

**Comment by jonashaag** [comment]
#459

**Comment by EduardJS** [comment]
Cool, thanks for letting me know. I thought that might be the case (regarding exclusion).

In thise case (aside mention ming it in the Docs), wouldn’t it be better for users to just leave those protocol dependent features disabled by default?

Have a great one!

**Comment by jonashaag** [comment]
I don’t think so! Non https production websites are almost extinct, at least actively maintained ones.

**Comment by Finesse** [comment]
Closed in favour of #459. We'll make a list of components that (do/don't) persist after changing the protocol.

**Comment by Valve** [comment]
@Finesse we don't support http in the pro version so the v3 will not support http. It will still work, of course, we'll just not be able to guarantee its work on http. It doesn't make sense to do anything related to fraud detection on plain HTTP websites. HTTPs is a pre-condition.
